hpedockerplugin/hpe/hpe_3par_mediator.py

```diff
@@ -0,0 +1,1032 @@
+    """
+
+    VERSION = "2.0.9"
+
+    def __init__(self, host_config, config):
+        self._host_config = host_config
+        self._config = config
+        self._client = None
+        self.client_version = None
+
+    @staticmethod
+    def no_client():
+        return hpe3parclient is None
+
+    def do_setup(self, timeout=30):
+
+        if self.no_client():
+            msg = _('You must install hpe3parclient before using the 3PAR '
+                    'driver. Run "pip install --upgrade python-3parclient" '
+                    'to upgrade the hpe3parclient.')
+            LOG.error(msg)
+            raise exception.HPE3ParInvalidClient(message=msg)
+
+        self.client_version = hpe3parclient.version_tuple
+        if self.client_version < MIN_CLIENT_VERSION:
+            msg = (_('Invalid hpe3parclient version found (%(found)s). '
+                     'Version %(minimum)s or greater required. Run "pip'
+                     ' install --upgrade python-3parclient" to upgrade'
+                     ' the hpe3parclient.') %
+                   {'found': '.'.join(map(six.text_type, self.client_version)),
+                    'minimum': '.'.join(map(six.text_type,
+                                            MIN_CLIENT_VERSION))})
+            LOG.error(msg)
+            raise exception.HPE3ParInvalidClient(message=msg)
+
+        try:
+            self._client = file_client.HPE3ParFilePersonaClient(
+                self._config.hpe3par_api_url)
+        except Exception as e:
+            msg = (_('Failed to connect to HPE 3PAR File Persona Client: %s') %
+                   six.text_type(e))
+            LOG.exception(msg)
+            raise exception.ShareBackendException(message=msg)
+
+        try:
+            ssh_kwargs = {}
+            if self._config.hpe3par_san_ssh_port:
+                ssh_kwargs['port'] = self._config.hpe3par_san_ssh_port
+            if self._config.ssh_conn_timeout:
+                ssh_kwargs['conn_timeout'] = self._config.ssh_conn_timeout
+            if self._config.hpe3par_san_private_key:
+                ssh_kwargs['privatekey'] = \
+                    self._config.hpe3par_san_private_key
+
+            self._client.setSSHOptions(
+                self._config.hpe3par_san_ip,
+                self._config.hpe3par_san_login,
+                self._config.hpe3par_san_password,
+                **ssh_kwargs
+            )
+
+        except Exception as e:
+            msg = (_('Failed to set SSH options for HPE 3PAR File Persona '
+                     'Client: %s') % six.text_type(e))
+            LOG.exception(msg)
+            raise exception.ShareBackendException(message=msg)
+
+        LOG.info("HPE3ParMediator %(version)s, "
+                 "hpe3parclient %(client_version)s",
+                 {"version": self.VERSION,
+                  "client_version": hpe3parclient.get_version_string()})
+
+        try:
+            wsapi_version = self._client.getWsApiVersion()['build']
```

Comment:
@imran-ansari will not it be good if we check here whether wsapi_version is less than required wsapi version.
If this is non supported version can we throw an error?

Reply:
WSAPI check is required for 3.3.1 Minimum since we are using some REST API's specific for 3.3.1